src/nut_agent.cc

```diff
@@ -224,13 +224,21 @@ void NUTAgent::advertisePhysics ()
             const auto &alarms = device.second.property ("ups.alarm");
             uint16_t bitfield = 0;
             int bit = 0;
+            int internal_failure_bit = 0;
             for (const auto &i : alarmsList) {
                 if (alarms.find(i) != std::string::npos) {
                     bitfield |= (1 << bit);
                     has_alarms = true;
                 }
+                if (i == "Internal UPS fault!") {
+                    internal_failure_bit = bit;
+                }
                 bit++;
             }
+            // TODO FIXME I hate this kind of fix, but it was to be quick and dirty
+            if (alarms.find("Internal failure!") != std::string::npos) {
+                bitfield |= (1 << internal_failure_bit);
+            }
```

Comment:
For clarity to future reviewers, we have a LUA rule-file that actually uses specific bit numbers to decide on criticality of the failure. Here we "remember" the number in array of the "Internal UPS fault" message (above in the loop), and alias it to other string(s... later?) with equivalent status reports that we have seen over other NUT drivers and protocols (e.g. NetXML vs SNMP).

Reply:
Better fix would be to remove bitfield, or to support vector of vectors for strings with the same meaning.